ClientApp/viking-vault/src/AttachCardModal.tsx

```diff
@@ -0,0 +1,215 @@
+import React from "react";
+import { constants } from "./ConstantVariables";
+import './AttachCardModal.css';
+
+const baseUrl = constants.baseUrl;
+var regex = /^([+-]?[1-9]\d*|0)$/;
+
+interface IModalProps {
+    open: boolean;
+    modalClose: any;
```

Comment:
I think that this would be better named `onModalClose` or `modalCloseHandler` as it works as an event